pkgs/build-support/go/module.nix

```
@@ -157,14 +143,10 @@ let
   }; in modArgs // (
       {
         outputHashMode = "recursive";
-      } // (if (vendorHashType == "sha256") then {
-        outputHashAlgo = "sha256";
-        outputHash = vendorSha256;
-      } else {
         outputHash = vendorHash;
-      }) // (lib.optionalAttrs (vendorHashType == "sri" && vendorHash == "") {
+      } // lib.optionalAttrs (args' ? vendorSha256 || vendorHash == "") {
         outputHashAlgo = "sha256";
```

```suggestion
        outputHashAlgo = if args' ? vendorSha256 || vendorHash == "" then "sha256" else null;
```

Passing `null` would make the structure simpler than using `optionalAttrs`.